### PR TITLE
Adding multithreading

### PR DESCRIPTION
**Pull request by wyxzou**
Adding multithreading and key listener event. Fixed some formatting as well.

### CODE REVIEW

File: code/controller/actuator.cpp [hunk 1]
```diff
@@ -22,152 +23,147 @@ Actuator::Actuator(const QString& serial_port,
...
-				throw "Action could not be completed"; // TODO: Might want to figure out a better way than throwing exceptions, revisit after adding concurrency
-			}
-
-			std::this_thread::sleep_for(sleep_step);
-		}
-	}
-	catch (...) {
-		throw std::current_exception();
-	}
+void Actuator::moveActuator(const unsigned char device, const int value, const int t) {
```
Comment (sadmansk) [inline code/controller/actuator.cpp]: Consider renaming `t` to `timer`? Or some other variable that would say more about what the parameter is used for. We should probably start using docstrings soon 😅

**Comment by sadmansk** [comment]
@wyxzou While you address that issue, I'll pull the code and test it out.

**Comment by sadmansk** [comment]
I will address rest of the issues myself.

**Comment by wyxzou** [comment]
Sorry for the late reply, I'll address the variable naming issue in my next commit. What do you mean my the rest of the issues?

**Comment by wyxzou** [comment]
Are you referring to the issue where program crashes on movement input?

**Comment by sadmansk** [comment]
@wyxzou just some more clean up, the source of the program crashing is actually known, it only happens if there arent any serial ports connected. So we are probably not catching the exception properly, or cleaning up the right way. Feel free to investigate nonetheless.

**Pull request merged**